### ISSUE TITLE
0.4.20 pactffi_mock_server_logs empty - mismatch in tracing library

### ISSUE THREAD

**Issue by YOU54F** [pact-foundation/pact-reference#434]
pactffi_mock_server_logs are empty, in 0.4.20 release.

Initially identified in pact-net's test suite when attempting to update to libpact_ffi 0.4.20. pact-net uses pactffi_log_to_buffer when it initialises, and has a test to check that pactffi_mock_server_logs actually returns logs (and it failed)

## Cause

I believe this is due to a mis-match in the logging libraries in `pact_models`, compared to the rest of the pact rust estate

- `tracing-0.1.37` https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_models/Cargo.toml#L48
- `tracing-subscriber-0.3.16` https://github.com/pact-foundation/pact-reference/blob/5699cab757e31875b74ea8ce98cd9821dc331421/rust/pact_models/Cargo.toml#L67C1-L67C93

All other crates

- https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_ffi/Cargo.toml#L49-L51
- https://github.com/pact-foundation/pact-reference/blob/5699cab757e31875b74ea8ce98cd9821dc331421/rust/pact_consumer/Cargo.toml#L39-L40
- https://github.com/pact-foundation/pact-reference/blob/5699cab757e31875b74ea8ce98cd9821dc331421/rust/pact_verifier/Cargo.toml#L48-L49
- https://github.com/pact-foundation/pact-reference/blob/5699cab757e31875b74ea8ce98cd9821dc331421/rust/pact_verifier_cli/Cargo.toml#L40-L43
- https://github.com/pact-foundation/pact-core-mock-server/blob/196cabbca4767df999e15f0aacb508549ba377a7/pact_mock_server/Cargo.toml#L43-L44
- https://github.com/pact-foundation/pact-plugins/blob/5a8e1df161fe7707146ea06b31588b35e9af2d08/drivers/rust/driver/Cargo.toml#L52-L53

## Reproductions

Added a test, which requires a buffer log sink. I believe the full test run sets a different global sink, so the test is ignored in the main run, and executed in isolation. It's not terribly pretty but it does the job.

✅ tag `libpact_ffi-v0.4.19`
 - Delta https://github.com/pact-foundation/pact-reference/compare/libpact_ffi-v0.4.19...YOU54F:pact-reference:issue/missing_logs_0.4.19
 - Passing Run https://github.com/YOU54F/pact-reference/actions/runs/9422764373/job/25959743140#step:6:121

❌ tag `libpact_ffi-v0.4.20`
 - Delta https://github.com/pact-foundation/pact-reference/compare/libpact_ffi-v0.4.20...YOU54F:pact-reference:issue/missing_logs_0.4.20
 - Failing Run https://github.com/YOU54F/pact-reference/actions/runs/9422936793/job/25960271309#step:6:66

## Resolution

Suggest

1. Update pact_models to match tracing libs
  1. Requires consuming in pact-plugin-driver & pact-core-mock-server
1. Pin logging libs explicitly with `version="=x.y.z"`, note the `=` sign in the version number. This forces pinned resolution

## Testing

1. Update pact_models to same logging libs
2. Update pact_mock_server to use updated pact_models
3. Update pact-plugin-driver, as it consumes pact_models, and are depended on by pact_mock_server
4. Update all the things in pact-ref to point to updated pact_mock_server and pact-plugin-driver

### pact-plugin-driver

- Delta https://github.com/pact-foundation/pact-plugins/compare/main...YOU54F:pact-plugins:fix/logging_min?expand=1
- Passing Run https://github.com/YOU54F/pact-plugins/actions/runs/9423940849/job/25963273163 (failures due to no token to publish to PactFlow)

### pact_mock_server

- Delta https://github.com/pact-foundation/pact-core-mock-server/compare/main...YOU54F:pact-core-mock-server:fix/logging_min?expand=1
- Passing Run https://github.com/YOU54F/pact-core-mock-server/actions/runs/9423753133 

### pact_reference

- Delta https://github.com/pact-foundation/pact-reference/compare/master...YOU54F:pact-reference:fix_logging?expand=1
- Passing run https://github.com/YOU54F/pact-reference/actions/runs/9423808069/job/25962909225

**Comment by simulified** [comment]
sup
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

**Comment by YOU54F** [comment]
should be fixed in 0.4.21, just awaiting release
